modules/invoicein/model/invoicein.class.php

```diff
@@ -351,24 +469,29 @@ public function journal() {
                         $VoucherH['voucher_Description']    = '';
                         $VoucherH['voucher_AccountPlanID']  = 0;
 
-                        $TotalPrice = round(($line->QuantityDelivered * $line->UnitCustPrice), 2);
-                        $TotalForeignPrice = round($line->ForeignAmount, 2);
+                        $query = "select sum(if(ChargeIndicator = 1, Amount, -Amount)) as sum from invoicelineallowancecharge where InvoiceType = 'in' and AllowanceChargeType = 'line' and InvoiceLineID = " . $line->LineID;
+                        $result = $_lib['storage']->get_row(array('query' => $query));
+                        $sum_line_allowance_charge = $result->sum;
+
+                        $TotalPrice = round(($line->QuantityDelivered * $line->UnitCustPrice + $sum_line_allowance_charge), 2);
+                        $TotalForeignPrice = round($line->ForeignAmount + $sum_line_allowance_charge, 2);
 
                         if($line->Vat > 0) {
-                            //#Add VAT to the price - since it is ex VAT
-                            //#print "$line->UnitCustPrice * (($line->Vat/100) +1)";
+                            //Add VAT to the price - since it is ex VAT
 
                             // $TotalPrice = round(($TotalPrice * (($line->Vat/100) +1)), 2);
                             // $TotalForeignPrice = round(($TotalForeignPrice * (($line->Vat/100) +1)), 2);
                             // We already have tax amount given to us by fakturabank.
                             // So why recalculate, and possibly make 0.01 error in recalculation?
                             // Just add them up and get the correct amount.
                             if ($VoucherH['voucher_ForeignCurrencyID'] != '') {
-                              $TotalForeignPrice = round(($TotalForeignPrice + $line->TaxAmount), 2);
-                              $TotalPrice = round(($TotalPrice + exchange::convertToLocal($VoucherH['voucher_ForeignCurrencyID'], $line->TaxAmount)), 2);
+                              $TotalForeignPrice = round(($TotalForeignPrice), 2);
+                              $TaxAmount = (!empty($line->TaxAmount)) ? $line->TaxAmount : $TotalForeignPrice * $line->Vat / 100.0;
+                              $TotalPrice = round(($TotalPrice + exchange::convertToLocal($VoucherH['voucher_ForeignCurrencyID'], $TaxAmount)), 2);
```

Comment:
Remove line 488.
Test for foreign invoice incoming, if it is bookkept correctly, if not this is to blame.

Possibly replace this:

```php
$TotalPrice = round(($TotalPrice + exchange::convertToLocal($VoucherH['voucher_ForeignCurrencyID'], $TaxAmount)), 2);
```

with:

```php
$TotalPrice = round((exchange::convertToLocal($VoucherH['voucher_ForeignCurrencyID'], $TotalForeignPrice) + exchange::convertToLocal($VoucherH['voucher_ForeignCurrencyID'], $TaxAmount)), 2);
```